### DISCUSSION TITLE
Does dynamic `import` have tree shaking?

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [webpack/webpack#15565] [Q&A]
Let's say I have a file called `util.js`:
```js
export const meta = {
  name: 'My Game',
  tags: ['Fun', 'Not Addictive']
}

export const game = {
  // The entire game (maybe 500 lines of code)
}
```

Then in my `index.js` file:
```js
async () => {
  const { name, tags } = (await import('./index.js')).meta
}
```

Would `game` also be downloaded / imported in the above file? I don't want `game` to be imported by the above code.

**Comment by vankop** [comment]
you can use `webpackExports` magic word to control this https://webpack.js.org/blog/2020-10-10-webpack-5-release/#general-tree-shaking-improvements

**Reply by ChocolateLoverRaj** [reply]
Will webpack do it automatically, like in my example?
